l10n_ch_account_statement_base_import/models/account_bank_statement_import.py

```diff
@@ -181,3 +181,18 @@ def _create_bank_statement(self, stmt_vals):
                 att = self.env['ir.attachment'].create(att_data)
 
         return statement_id, notifs
+
+    @api.model
+    def _find_bank_account_id(self, account_number):
+        """ Override to find Postfinance account Given IBAN number """
+        bank_account_id = super(AccountBankStatementImport, self).\
+            _find_bank_account_id(account_number)
+        if not bank_account_id and account_number and len(
+                account_number) == 21:
+            pf_formated_acc_number = account_number[9:].lstrip('0')
+            bank_account_ids = self.env['res.partner.bank'].search(
+                [('sanitized_acc_number', '=', pf_formated_acc_number)],
+                limit=1)
+            bank_account_id = bank_account_ids[0].id if bank_account_ids\
```

Comment:
Simply do `bank_account_id = bank_account.id`

If is a record set `rec` has 0 elements, `rec.id` returns `False`